### PR TITLE
Update json, msgpack, yajl-ruby, oj, td-client and td dependency versions and add google-protobuf version pin to be compatible with Ruby 2.4.

### PR DESCRIPTION
**Pull request by qingling128**

### CODE REVIEW

**Comment by qingling128** [comment]
This has been tested manually (on a Debian 8 machine).

File: core_gems.rb [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 download "http_parser.rb", "0.6.0"
-download "yajl-ruby", "1.2.1"
+download "yajl-ruby", "1.3.1"
 download "sigdump", "0.2.4"
 download "thread_safe", "0.3.5"
-download "oj", "2.15.1"
+download "oj", "3.3.10"
 download "tzinfo", "1.2.2"
 download "tzinfo-data", "1.2016.5"
+download "google-protobuf", "3.5.1"
```
Comment (igorpeshansky) [inline core_gems.rb]: This seems new. Any reason we're doing this explicitly, rather than letting another gem pull it in as a dependency?

Comment (qingling128) [reply core_gems.rb]: I had to explicitly pin `google-protobuf` to `3.5.1` because otherwise it pulls in the latest `3.5.1.1` which is somehow broken.

Comment (igorpeshansky) [reply core_gems.rb]: Let's update the PR description, then.

Comment (qingling128) [reply core_gems.rb]: Updated. Also opened a ticket to report the issue with `3.5.1.1`: https://github.com/google/protobuf/issues/4210.

**Comment by igorpeshansky** [comment]
:+1:

File: core_gems.rb (hunk 1 above)
Comment (igorpeshansky) [reply core_gems.rb]: :+1:

**Pull request merged**